src/http_handlers.cpp

```diff
@@ -1100,12 +1100,22 @@ void ImpuRegDataTask::put_in_cache()
         // LCOV_EXCL_START - This is essentially tested in the PPR UTs
         TRC_ERROR("No SIP URI in Implicit Registration Set");
         SAS::Event event(this->trail(), SASEvent::NO_SIP_URI_IN_IRS, 0);
-        event.add_compressed_param(_xml, &SASEvent::PROFILE_SERVICE_PROFILE);
+        event.add_compressed_param(_irs->get_ims_sub_xml(), &SASEvent::PROFILE_SERVICE_PROFILE);
         SAS::report_event(event);
         // LCOV_EXCL_STOP
       }
     }
 
+    SAS::Event event(this->trail(), SASEvent::CACHE_PUT_REG_DATA, 0);
```

Comment:
I think I'd add a block scope here (`{` and `}`), which will mean we scope the variables used to send the SAS event.

Alternatively, I'd have wrapped it up in a function.